lib/internal/test_runner/mock/mock_timers.js

```diff
@@ -98,7 +116,80 @@ class MockTimers {
     this.#executionQueue.removeAt(position);
   }
 
-  async * #setIntervalPromisified(interval, startTime, options) {
+  #createDate() {
+    kMock ??= Symbol('MockTimers');
+    /**
+     * Function to mock the Date constructor, treats cases as per ECMA-262
+     * and returns a Date object with a mocked implementation
+     * @typedef {Date} MockDate
+     * @returns {MockDate} a mocked Date object
+     */
+    function MockDate(year, month, date, hours, minutes, seconds, ms) {
+      const mockTimersSource = MockDate[kMock];
+      const nativeDate = mockTimersSource.#RealDate;
+      // As of the fake-timers implementation for Sinon
+      // ref https://github.com/sinonjs/fake-timers/blob/a4c757f80840829e45e0852ea1b17d87a998388e/src/fake-timers-src.js#L456
+      // This covers the Date constructor called as a function ref.
+      // ECMA-262 Edition 5.1 section 15.9.2.
+      // and ECMA-262 Edition 14 Section 21.4.2.1
+      if (!(this instanceof MockDate)) {
```

Comment:
instanceof is forgeable and breakable, this should use the DatePrototypeGetTime approach.

Reply:
I ran into some issues when using only that function because when called directly the `this` property is `null`, so I implemented a check for it but I'm not sure if it should cover all the cases, can you give some insight on [this](https://github.com/nodejs/node/pull/48638/files#diff-f55a12a3f70eb028a92d8d076db8456c693b72bf669c9c6856bdaec496a3896aR64) line?